### PR TITLE
fixes a problem with the --savetiddler command, 

### PR DESCRIPTION
**Pull request by pmario**
it the tiddler doesn't exist and adds a propper error message

to reproduce 

```
node $(TIDDLYWIKI5_DIR)/tiddlywiki.js \
    editions \
    --verbose \
    --rendertiddler $:/core/save/all $(DEV_DIR)/index.html text/plain \
    --savetiddler $:/XXXXXXXXXXxxfavicon.ico $(DEV_DIR)/favicon.ico \
    || exit 1
```

The XXXXfavicon tiddler will cause a stack trace error, which doesn't help. I did add a check and a propper message, that tells the user, what to do .

### CODE REVIEW

**Comment by pmario** [comment]
@Jermolene  
This problem hit me again, because of a typo. Is there a chance, that it gets merged if the texts are translatable?

**Comment by Jermolene** [comment]
Hi @pmario I realised that there are already en-GB error messages embedded in some of the existing commands, so I'll go ahead and merge, and then make a couple of tweaks.

**Comment by Jermolene** [comment]
@pmario this pull request is no longer merge-able, could you kindly adjust it? I've got a couple of minor code comments too.

File: core/modules/commands/rendertiddler.js [hunk 1]
```diff
@@ -32,10 +32,20 @@ Command.prototype.execute = function() {
...
-		type = this.params[2] || "text/html";
-	fs.writeFile(filename,this.commander.wiki.renderTiddler(type,title),"utf8",function(err) {
-		self.callback(err);
-	});
+		type = this.params[2] || "text/html",
+		tiddler;
+
+	tiddler = this.commander.wiki.getTiddler(title);
+	// Check if tiddler exists. Common user error is a typo
+	if(tiddler) {
```
Comment (Jermolene) [inline core/modules/commands/rendertiddler.js]: Here I think you could use `if(this.command.wiki.tiddlerExists(title)) {`

File: core/modules/commands/rendertiddler.js [hunk 2]
```diff
@@ -32,10 +32,20 @@ Command.prototype.execute = function() {
...
+		fs.writeFile(filename,this.commander.wiki.renderTiddler(type,title),"utf8",function(err) {
+			self.callback(err);
+		});
+	} else {
+		return	"\n-----> Command: --" + exports.info.name + " " + title +
```
Comment (Jermolene) [inline core/modules/commands/rendertiddler.js]: The existing error messages are much terser than this, and we don't echo the command name (the user should use `--verbose` to get that). Perhaps just:

```
return "The tiddler '" + title + "' does not exist";
```

File: core/modules/commands/savetiddler.js [hunk 3]
```diff
@@ -32,12 +32,24 @@ Command.prototype.execute = function() {
 		path = require("path"),
 		title = this.params[0],
 		filename = this.params[1],
-		tiddler = this.commander.wiki.getTiddler(title),
-		type = tiddler.fields.type || "text/vnd.tiddlywiki",
+		tiddler,
+		type,
+		contentTypeInfo;
+
```
Comment (Jermolene) [inline core/modules/commands/savetiddler.js]: Mind the whitespace!

**Comment by Jermolene** [comment]
Hi @pmario I think the only outstanding issue is the extraneous text in the error messages, if you're still interested in getting this merged?

**Comment by pmario** [comment]
Yea. The error message is used by a missing file. .... I'll create a new PR, which closes this one.

**Comment by pmario** [comment]
--rendertiddler XX commands are deprecated. So close it

**Pull request closed without merging**